test/com/amazon/ion/impl/bin/IonEncoder_1_1Test.java

```diff
@@ -455,6 +457,127 @@ public void testWriteTimestampValueForNullTimestamp() {
         Assertions.assertEquals(2, numBytes);
     }
 
+    @ParameterizedTest
+    @CsvSource({
+            "'', 80",
+            "'a', 81 61",
+            "'ab', 82 61 62",
+            "'abc', 83 61 62 63",
+            "'fourteen bytes', 8E 66 6F 75 72 74 65 65 6E 20 62 79 74 65 73",
+            "'this has sixteen', F8 21 74 68 69 73 20 68 61 73 20 73 69 78 74 65 65 6E",
+            "'variable length encoding', F8 31 76 61 72 69 61 62 6C 65 20 6C 65 6E 67 74 68 20 65 6E 63 6F 64 69 6E 67",
```

Comment:
For the text tests, it would be kind of nice to allow hex or ASCII characters. Maybe it would work if, after splitting on spaces, if a string has length 1 it's replaced with its ASCII value. Then you could have

```
"'fourteen bytes', 8E f o u r t e e n 20 b y t e s",
```
Similar to how you in Java you can do stuff like `BitUtils.bytes(0x8E, 'f', 'o', 'u', 'r' ...)`, for readability.

Reply:
I tried it, but it was starting to become a very invasive change.